staticaddr/deposit/actions.go

```diff
@@ -146,3 +146,19 @@ func (f *FSM) SweptExpiredDepositAction(_ fsm.EventContext) fsm.EventType {
 
 	return fsm.NoOp
 }
+
+// WithdrawnDepositAction is the final action after a withdrawal. It signals to
+// the manager that the deposit has been swept and the FSM can be removed. It
+// also ends the state machine main loop by cancelling its context.
+func (f *FSM) WithdrawnDepositAction(_ fsm.EventContext) fsm.EventType {
+	select {
+	case <-f.ctx.Done():
+		return fsm.OnError
+
+	default:
+		f.finalizedDepositChan <- f.deposit.OutPoint
+		f.ctx.Done()
```

Comment:
`f.ctx.Done()` does not close the context. The description of the function says "ends the state machine main loop by cancelling its context", but this call just returns the context, it is a no-op.

Reply:
Good catch, removed it. After this call the FSM is in a final state.